src/altinn-app-frontend/src/components/base/MapComponent.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+
+import type { IComponentProps } from '..';
+import type { Location, MapLayer } from '@altinn/altinn-design-system';
+import { Map } from '@altinn/altinn-design-system';
+
+import './MapComponent.css';
+import '../../styles/shared.css';
+import {
+  getLanguageFromKey,
+  getParsedLanguageFromKey,
+} from 'altinn-shared/utils';
+import { Typography } from '@material-ui/core';
+
+export interface IMapComponentProps extends IComponentProps {
+  layers?: MapLayer[];
+  centerLocation?: Location;
+  zoom?: number;
+}
+
+export function MapComponent({
+  formData,
+  handleDataChange,
+  language,
+  readOnly,
+  layers,
+  centerLocation,
+  zoom,
+}: IMapComponentProps) {
+  const location = formData.simpleBinding
+    ? parseLocation(formData.simpleBinding)
+    : undefined;
+
+  const footerText = location
+    ? getParsedLanguageFromKey('map_component.selectedLocation', language, [
+        location.latitude,
+        location.longitude,
+      ])
+    : getLanguageFromKey('map_component.noSelectedLocation', language);
+
+  const handleMapClicked = (location: Location) => {
+    handleDataChange(`${location.latitude},${location.longitude}`);
+  };
+
+  return (
+    <>
+      <Map
+        layers={layers}
+        centerLocation={location || centerLocation}
+        zoom={location ? 16 : zoom}
+        markerLocation={location}
+        readOnly={readOnly}
+        onClick={handleMapClicked}
+      />
+      <Typography>{footerText}</Typography>
```

Comment:
You can style the component by [overriding with your own css classes](https://mui.com/material-ui/customization/how-to-customize/#overriding-styles-with-class-names). This can be done directly in the code, I have added some example code in the review comments for `MapComponent.tsx` that you can have a look at 🙂 Check out f.ex. the ImageComponent or our components in the `shared` folder for more examples